src/sparkx/Loader/JetscapeLoader.py

```diff
@@ -0,0 +1,453 @@
+
+        Parameters
+        ----------
+        None
+
+        Raises
+        ------
+        None
+
+        Returns
+        -------
+        None
+        """
+        with open(self.PATH_JETSCAPE_, 'r') as jetscape_file:
+            event_output = []
+    
+            while True:
+                line = jetscape_file.readline()
+                if not line:
+                    break
+                elif '#' in line and self.particle_type_defining_string_ in line:
+                    line_str = line.replace('\n','').replace('\t',' ').split(' ')
+                    event = line_str[2]
+                    num_output = line_str[8]
+                    event_output.append([event, num_output])
+                else:
+                    continue
+
+        self.num_output_per_event_ = np.asarray(event_output, dtype=np.int32)
+        self.num_events_ = len(event_output)
+
+    def get_last_line(self,file_path):
+        """
+        Returns the last line of a file.
+
+        Parameters
+        ----------
+        file_path : str
+            The path to the file.
+
+        Returns
+        -------
+        str
+            The last line of the file, stripped of leading and trailing whitespace.
+        """
+        with open(file_path, 'rb') as file:
+            file.seek(-2, 2)
+            while file.read(1) != b'\n':
+                file.seek(-2, 1)
+            last_line = file.readline().decode().strip()
+            return last_line
+
+    def get_sigmaGen(self):
+        """
+        Retrieves the sigmaGen values with error from the last line of a file.
+
+        Returns
+        -------
+        tuple
+            A tuple containing the first and second sigmaGen values as floats.
+        """
+        last_line = self.get_last_line(self.PATH_JETSCAPE_)
+        words = last_line.split()
+        numbers = []
+
+        for word in words:
+            try:
+                number = float(word)
+                numbers.append(number)
+                if len(numbers) == 2:
+                    break
+            except ValueError:
+                continue
+
+        return (numbers[0],numbers[1])
+    
+    def get_particle_type_defining_string(self):
+        """
+        Returns the string which definies the particle type in the Jetscape file.
+
+        Returns
+        -------
+        string
+            The particle type defining string.
+        """
+        return self.particle_type_defining_string_
+    
+    def get_particle_type(self):
```

Comment:
I just realized that there is no specific test for this function.

Reply:
Not exactly no. This is kinda tested by the Jetscape tests, which are now more like integration tests. We could of course also add specific tests for the Loaders.